src/containers/PerfherderGraphContainer/index.jsx

```diff
@@ -22,10 +23,25 @@ const styles = () => ({
 class PerfherderGraphContainer extends Component {
   state = {
     data: null,
+    tooltipModel: null,
+    canvas: null,
   };
 
   async componentDidMount() {
-    this.fetchSetData(this.props);
+    await this.fetchSetData(this.props);
+
+    const self = this;
+
+    this.setState(prevState => {
+      const { options } = prevState;
+
+      options.tooltips.custom = function custom(tooltipModel) {
+        // eslint-disable-next-line no-underscore-dangle
+        self.setState({ tooltipModel, canvas: this._chart.canvas });
```

Comment:
One more thing, now that this is inlined, I can see that we only need to set ` canvas: this._chart.canvas` once.  Can it be done outside of `custom()`, just above?

Reply:
It's not possible. Again, because this._chart only exists inside the custom function. 

The custom function is not being executed right away, it will be executed when the tooltip pops up by ChartJS, and what we are doing is creating the custom function here so we can bind those two properties (`this._chart`, `tooltipModel`) to the `PerfherderGraphContainer` and do magic as you established before.

I know it doesn't look good, but for now it's the cleanest way I can imagine to bind chartjs behaviour with our react components. I'm still not giving up, and if I come with something else in the future, I'll let you know :)

Reply:
Also, I like your idea about using HOC. I started yesterday already reading about them and yup I'll try to use them in the next PR.